### PR TITLE
add current scope variables when evaluating an expression

### LINKED ISSUE #2242: [🐛 BUG] Scope issue with lambda function
### What went wrong? 🤔

There are issues using the lambda functions with multiple pages.

```
C:\Users\jacta\.conda\envs\test\Lib\site-packages\taipy\gui\utils\_evaluator.py:369: TaipyGuiWarning: Exception raised evaluating __lambda_2466438897376_1_TPMDL_0(x):

name 'create_chart' is not defined
  _warn(f"Exception raised evaluating {expr_string}", e)

```

(Note that this issue was first seen when trying to have multiple states for an application; this might not be the same issue)

### Expected Behavior

This should work fine like the comments in the code example.

### Steps to Reproduce Issue

*app.py*

```python
from taipy import Gui
from page_py import page # if you add 'create_chart' in the import, then this works!

pages = {"/": "App", "page": page}

Gui(pages=pages).run()
```

*page_py.py*
```python
import taipy.gui.builder as tgb
import pandas as pd
import plotly.express as px

x = 0


def create_chart(x):
    print("Hello")
    df = pd.DataFrame(
        {
            "Fruit": ["Apples", "Oranges", "Bananas", "Apples", "Oranges", "Bananas"],
            "Amount": [4, 1, 2, 2, 4, 5],
            "City": ["SF", "SF", "SF", "Montreal", "Montreal", "Montreal"],
        }
    )
    fig = px.bar(df, x="Fruit", y="Amount", color="City", barmode="group")
    fig.show()
    return fig


with tgb.Page() as page:
    tgb.text("Hello World")
    tgb.slider("{x}")
    tgb.chart(figure=lambda x: create_chart(x))

    # If you use the Markdown syntax, there are no issues. This works
    # tgb.chart(figure="{create_chart(x)}")
```` 

### Browsers

Chrome

### OS

Windows

### Version of Taipy

4.0.1

### Acceptance Criteria

- [ ] A unit test reproducing the bug is added.
- [ ] Any new code is covered by a unit tested.
- [ ] Check code coverage is at least 90%.
- [ ] The bug reporter validated the fix.
- [ ] Related issue(s) in taipy-doc are created for documentation and Release Notes are updated.

### Code of Conduct

- [X] I have checked the [existing issues](https://github.com/Avaiga/taipy/issues?q=is%3Aissue+).
- [ ] I am willing to work on this issue (optional)

### PR DESCRIPTION
**Pull request by FredLL-Avaiga**
resolves #2242

## What type of PR is this? (check all applicable)

- [ ] Refactor
- [ ] Feature
- [x] Bug Fix
- [ ] Optimization
- [ ] Documentation Update

## Description
When using lambda in another page context, the locals are not resolved

## Related Tickets & Documents
- Closes #2242 

## How to reproduce the issue

see issue

## Checklist

- [x] Does this solution meet the acceptance criteria of the related issue?
- [ ] Is the related issue checklist completed?
- [ ] Does this PR adds unit tests for the developed code? If not, why?
- [ ] End-to-End tests have been added or updated?
- [ ] Was the documentation updated, or a dedicated issue for documentation created? (If applicable)
- [ ] Is the release notes updated? (If applicable)

### CODE REVIEW

File: tests/gui/gui_specific/test_evaluator.py [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+#
+# Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+# an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+# specific language governing permissions and limitations under the License.
+
+from taipy.gui import Gui
+from taipy.gui.utils._evaluator import _Evaluator
+
+
+def _create_chart(x):
```
Comment (FabienLelaquais) [inline tests/gui/gui_specific/test_evaluator.py]: 'create_chart()', really? What about '_identity()' or something less significant?

**Review by FabienLelaquais — DISMISSED** [review]
👌

File: tests/gui/gui_specific/test_evaluator.py (hunk 1 above)
Comment (FredLL-Avaiga) [reply tests/gui/gui_specific/test_evaluator.py]: ok

**Review by dinhlongviolin1 — APPROVED** [review]

**Review by FabienLelaquais — APPROVED** [review]
👍

**Pull request merged**